crypto/src/main/kotlin/net/corda/v5/crypto/merkle/MerkleTreeHashDigestFactory.kt

```diff
@@ -0,0 +1,14 @@
+package net.corda.v5.crypto.merkle
+
+import net.corda.v5.crypto.DigestAlgorithmName
+
+/**
+ * Different use cases require different [MerkleTree] calculations.
+ * [MerkleTreeHashDigestFactory]s let us specify such implementations.
+ *
+ *  @property digestAlgorithmName Specifies the digest algorithm.
+ */
+
+interface MerkleTreeHashDigestFactory {
```

Comment:
Can we rename to MerkleTreeHashDigest

Reply:
Why though? I'm a bit confused about the purpose of the interface now.

Reply:
The implementation of that interface would be a proxy to the MerkleTreeHashDigestProvider. That's again separation of concern and controlling what is actually available to the CorDApp developer, the contract of that interface suites the app developer as the actual implementation may have stuff which is of no interest to the app developer. If you look the methods are actually different